core/src/main/java/org/ehcache/core/StatusTransitioner.java

```diff
@@ -218,7 +218,13 @@ public void succeeded() {
     }
 
     public StateTransitionException failed(Throwable t) {
+      if (st.done()) {
```

Comment:
This condition feels wrong - it probably has to be linked with the `Throwable` being null or not.